### PR TITLE
Add check for missing cookieKey, generate one if not specified

### LINKED ISSUE #190: Null Pointer Exception if you start without a config file
Exception in thread "main" java.lang.IllegalArgumentException: (was java.lang.NullPointerException) (through reference chain: keywhiz.KeywhizConfig["cookieKey"])

We should print an error message, not just NPE.

I'm not sure if this is something specific to the cookieKey, if so, Issue #180 would fix it.

### PR DESCRIPTION
**Pull request by worldwise001**
Should fix #190 and #180 .

r: @mcpherrinm @csstaub @alokmenghrajani

### CODE REVIEW

**Comment by mcpherrinm** [comment]
Could we deterministically generate one, perhaps like we do the per-secret keys?

Randomly generating one is going to lead to confusion and sadness as your cookies are only valid on the server that generated it.

**Comment by worldwise001** [comment]
Yeah we could. What would we use as a stable value to deterministically generate one reliably on multiple servers?

I guess the other consideration is that if you're going to configure more than one, maybe you will need to specify one manually.

**Comment by mcpherrinm** [comment]
We have a master key that we derive keys per-secret -- we could derive a key for cookies too.
However, we'd have to be careful whatever we use for derivation doesn't collide with the name of a secret.

see  https://github.com/square/keywhiz/blob/master/server/src/main/java/keywhiz/service/crypto/ContentCryptographer.java

File: server/src/main/java/keywhiz/ServiceModule.java [hunk 1]
```diff
@@ -75,6 +87,14 @@ public ServiceModule(KeywhizConfig config, Environment environment) {
     bind(KeywhizConfig.class).toInstance(config);
   }
 
+  private String generateCookieKey() {
```
Comment (csstaub) [inline server/src/main/java/keywhiz/ServiceModule.java]: This means the cookie key won't work across hosts, and it'll also get wiped on restart and everyone ends up being logged out?

**Comment by worldwise001** [comment]
We can derive off the master key. I'll submit an update.

**Comment by worldwise001** [comment]
Whoops I completely forgot about this. I'll see if I can get this updated today.

**Comment by mcpherrinm** [comment]
Abandoned PR

**Pull request closed without merging**